components-java-sdk/src/main/java/io/dapr/components/domain/state/StateStore.java

```diff
@@ -16,13 +16,77 @@
 import io.dapr.components.aspects.AdvertisesFeatures;
 import io.dapr.components.aspects.InitializableWithProperties;
 import io.dapr.components.aspects.Pingable;
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
+import java.util.List;
 
+/**
+ * Represents a state store.
+ *
+ * <p>Default implementations are provided for bulk operations. They relly on their
+ * single-item variants and can be made more efficient if your underlying state store
+ * provides native support for bulk operations.</p>
+ */
 public interface StateStore extends InitializableWithProperties, AdvertisesFeatures, Pingable {
+  /**
+   * Get data from the given key.
+   *
+   * @param getRequest Request specifying what to retrieve.
+   * @return A {@link GetResponse} Mono containing the contents associated with the key or
+   *         representing an error.
+   */
   Mono<GetResponse> get(GetRequest getRequest);
 
+  /**
+   * Deletes the specified key from the state store.
+   * @param deleteRequest Request specifying what to delete.
+   * @return An empty Mono representing success or error.
+   */
   Mono<Void> delete(DeleteRequest deleteRequest);
 
+  /**
+   * Sets the value of the specified key.
+   *
+   * @param setRequest Request specifying what key to set and its contents.
+   * @return An empty Mono representing success or error.
+   */
   Mono<Void> set(SetRequest setRequest);
+
+  /**
+   * Retrieves many keys at once.
+   *
+   * @param getRequests list of GetRequest to be performed in bulk.
+   * @return A Flux of GetResponses, one for each requested key.
+   */
+  default Flux<BulkGetStateItem> bulkGet(List<GetRequest> getRequests) {
```

Comment:
While I'm not a fan of abstract types because they tend to obscure the methods that need to be implemented for a given feature, default interfaces like this feels like a good compromise.  I'll have to see if we can (and it makes sense) to do something similar with the newer default interface methods in C#/.NET.

Reply:
Yeah. Those default methods are a nifty feature from Java. Other languages have come with mix-in like solutions that would accomplish something really similar but that the closest regular java has to this.